### ISSUE TITLE
Question about the GNSS payload

### ISSUE THREAD

**Issue by ZenoMaverick** [Lora-net/SWSD004#2]
Hi, sorry this may be a dumb question but I've been struggling to understand this particular part in the[ GNSS scan information](https://github.com/Lora-net/SWSD004/tree/master/apps/examples/geolocation_gnss#execute-and-send-the-assisted-multiframe-gnss-scan). Can I check how is this long hex part meant to be read? Is this the NAV message?

`scan[0][1335796859] (10 SV): 8247EA010A01567378AFF1C58A85C9DAAC54199BD49A4C2245D6E833409C1A89B0020B00`

I checked [_gnss_helpers.c_](https://github.com/Lora-net/SWSD004/blob/390c7b3f321941057b5c60e9e25c5b2b47475773/geolocation_middleware/gnss/src/gnss_helpers.c) but I'm a little confused whether this contains satellite related information at all.
Could I kindly check this? Thanks!

**Comment by smtcbboulet** [comment]
Hello,

This payload : 8247EA010A01567378AFF1C58A85C9DAAC54199BD49A4C2245D6E833409C1A89B0020B00 is the NAV message. it contains the information related to the satellites yes. 

Benjamin

**Comment by ZenoMaverick** [comment]
Hi Benjamin, thanks so much for the clarifications. I've tried it out the GNSS example and now it makes more sense. 👍  

**Could I check two more things about the NAV message?**
1. Are you able to share how this NAV message is created? I checked the [Semtech Lora Cloud documentation](https://www.loracloud.com/documentation/geolocation?url=gnss.html) but there's not much information how it's formatted after doing the GNSS scanning.

2. It seems like this NAV message can currently only be decrypted by Lora Cloud and there's no way to implement a custom GNSS solver. Is this correct or are we able to do so?

Thanks for your help!
